src/components/Dialogs/Dialog/Dialog.tsx

```diff
@@ -18,42 +21,66 @@ export interface DialogProps extends ModalProps {
     /** Dialog title */
     title?: string;
     /** Properties of the accept button */
-    acceptButton: ButtonProps;
+    acceptButton?: ButtonProps;
     /** Properties of the cancel button */
     cancelButton?: ButtonProps;
+    /** the context of the buttons in the footer */
+    variant?: Omit<ButtonContext, 'secondary'>;
+    /** closes the dialog */
+    onClose: () => void;
 }
 
 const { block, elem } = bem('Dialog', styles);
 
 export const Dialog: React.FC<DialogProps> = (props) => {
-    const { isOpen, children, title, acceptButton, cancelButton, ...rest } = props;
+    const {
+        isOpen,
+        children,
+        title,
+        acceptButton,
+        cancelButton,
+        variant = 'primary',
+        onClose,
+        ...rest
+    } = props;
 
+    const { ref, height } = useResizeObserver<HTMLDivElement>();
+    const showButtons = !!acceptButton;
     const isConfirm = !!cancelButton;
 
     return (
-        <Modal {...rest} {...block(props)} isOpen={isOpen}>
-            <div {...elem('content', props)} role="alert">
+        <Modal {...rest} {...block('dialog')} isOpen={isOpen}>
+            <div {...elem('header')} role="alert">
                 {!!title && (
                     <Heading level="h2" {...elem('title', props)}>
                         {title}
                     </Heading>
                 )}
+                <MdClose onClick={onClose} data-testid="close-icon" />
+            </div>
+            <div ref={ref} {...elem('content')}>
                 {children}
             </div>
-            <div {...elem('actions', props)}>
-                {isConfirm && (
+            {showButtons && (
+                <div {...elem('actions', { ...props, borderTop: (height ?? 0) === 528 })}>
                     <Button
-                        {...elem('cancel', props)}
-                        onClick={cancelButton.onClick}
-                        variant="ghost"
+                        onClick={acceptButton.onClick}
+                        context={variant === 'primary' ? 'primary' : 'critical'}
+                        aria-label="Confirm OK"
                     >
-                        {cancelButton.label}
+                        {acceptButton.label}
                     </Button>
-                )}
-                <Button onClick={acceptButton.onClick} context="primary">
-                    {acceptButton.label}
-                </Button>
-            </div>
+                    {isConfirm && (
```

Comment:
I'm not sure how this prop makes sense in the current context. Shouldn't the condition be if there is a cancel button present or not?

Reply:
isConfirm is not a prop. It's calculated from cancelButton (line 49). It was like this so I kept it..

Reply:
Please confirm with design team first. Looking at the variations both buttons should always be rendered together, no need for isConfirm. If it is to be rendered together, the props needs to be changed and a single object to contain both button props.

Reply:
This introduces a breaking change, please mark the commit as breaking change

Reply:
Even if the prop was named this way before, I don't see the reason to keep this naming if it doesn't make sense in the current context anymore. This will make the code confusing and hard to understand in the future. In general, I'd rather it gets renamed, especially if this is only internal to this file, it is very easy to do. 
However looking at the code as it looks now, it doesn't even seem to make a lot of sense to me to have this as a variable. Maybe it is easier to just use `!!cancelButton` in place, same as you do with the `!!title` above.